src/DataStructures/IndexIterator.hpp

```diff
@@ -35,7 +35,7 @@ class IndexIterator {
   /// \endcond
 
   /// Returns false if the end of the Index iteration is reached
-  operator bool() const noexcept { return valid_; }
+  operator bool() const noexcept { return valid_; }  // NOLINT
```

Comment:
I'm guessing that this was complaining that the operator isn't explicit.  (I don't see anything else interesting here.)  Rather than silencing the warning, I think we want these to be explicit.